solr/core/src/java/org/apache/solr/client/solrj/embedded/JettySolrRunner.java

```diff
@@ -304,7 +304,11 @@ private void init(int port) {
       ServerConnector connector;
       if (sslcontext != null) {
         configuration.setSecureScheme("https");
-        configuration.addCustomizer(new SecureRequestCustomizer());
+        SecureRequestCustomizer customizer = new SecureRequestCustomizer(false);
+        sslcontext.setSniRequired(false);
+        customizer.setSniHostCheck(false);
```

Comment:
SNI is server name indication and as far as I've seen only ever required if you have multiple websites behind a single server - and the server needs to figure out which website for which TLS certificate. Basically the ability to host multiple sites with multiple TLS certificates.

JettySolrRunner either in tests or as an embedded server doesn't need to support SNI so makes SNI not required and doesn't do the SNI check.

Reply:
It would be extremely unlikely that Solr would ever have more than one certificate chain installed, so I don't think we need to worry about SNI.  We actively discourage running more than one webapp in the same container as Solr, which I think would be about the only time you'd actually want SNI.  If somebody wants to get that fancy, they are on their own for getting Jetty to support what they need.

I'd really like to get to the point where we embed Jetty and manage the whole configuration ourselves.  Maybe have the Jetty configuration in solr.xml.  Then I think we could utilize bouncycastle to allow users to supply PEM files for TLS, to be converted on the fly at startup into what Java wants.  We are already including BC dependencies in Solr.